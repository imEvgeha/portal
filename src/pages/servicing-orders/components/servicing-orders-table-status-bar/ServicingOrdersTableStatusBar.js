import React from 'react';
import PropTypes from 'prop-types';
import './ServicingOrdersTableStatusBar.scss';


const ServicingOrdersTableStatusBar = ({statusBarInfo}) => {
    const baseClassName = 'nexus-c-servicing-orders-table-status-bar';

    const {totalRows, selectedRows} = statusBarInfo || {};

    return (
        <div className={baseClassName}>
            <span className={`${baseClassName}__description`}>
                Rows: <span className={`${baseClassName}__value`}>{totalRows}</span>
            </span>
            {selectedRows > 0 && (
                <span className={`${baseClassName}__description`}>
                    Selected: <span className={`${baseClassName}__value`}>{selectedRows}</span>
                </span>
            )}
        </div>
    );
};

ServicingOrdersTableStatusBar.propTypes = {
    statusBarInfo: PropTypes.string,
};

ServicingOrdersTableStatusBar.defaultProps = {
    statusBarInfo: '',
};

export default ServicingOrdersTableStatusBar;
