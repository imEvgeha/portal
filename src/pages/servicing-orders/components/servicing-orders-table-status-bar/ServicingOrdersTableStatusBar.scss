.nexus-c-servicing-orders-table-status-bar {
  position: absolute;
  top: calc(100% - 1px);
  height: 35px;
  width: 100%;
  padding: 0 16px;
  display: flex;
  align-items: center;
  background-color: white;
  border: solid 1px #bdc3c7;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  font-size: 12px;

  &__info {
    &:not(:last-of-type) {
      margin-right: 12px;
    }

    &--bold {
      font-weight: bold;
    }
  }
}
