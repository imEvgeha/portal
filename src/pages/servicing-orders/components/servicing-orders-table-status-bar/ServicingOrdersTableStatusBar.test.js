import React from 'react';
import {shallow} from 'enzyme';
import ServicingOrdersTableStatusBar from './ServicingOrdersTableStatusBar';

describe('ServicingOrdersTableStatusBar', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<ServicingOrdersTableStatusBar statusBarInfo={{totalRows: 5, selectedRows: 3}} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should show the selected row label if there are selected rows', () => {
        const wrapper = shallow(<ServicingOrdersTableStatusBar statusBarInfo={{totalRows: 5, selectedRows: 3}} />);
        const selectedRowsText = wrapper.find('.nexus-c-servicing-orders-table-status-bar__description').at(1).text();

        expect(selectedRowsText).toContain('Selected');
    });

    it('should not show the selected row label if there are no selected rows', () => {
        const wrapper = shallow(<ServicingOrdersTableStatusBar statusBarInfo={{totalRows: 5, selectedRows: 0}} />);
        const selectedRowsNode = wrapper.find('.nexus-c-servicing-orders-table-status-bar__description').at(1).length;

        expect(selectedRowsNode).toBeFalsy();
    });

    it('should show the correct number of total rows', () => {
        const wrapper = shallow(<ServicingOrdersTableStatusBar statusBarInfo={{totalRows: 55, selectedRows: 0}} />);
        const totalRowsText = wrapper.find('.nexus-c-servicing-orders-table-status-bar__description').at(0).text();

        expect(totalRowsText).toContain('55');
    });
});
