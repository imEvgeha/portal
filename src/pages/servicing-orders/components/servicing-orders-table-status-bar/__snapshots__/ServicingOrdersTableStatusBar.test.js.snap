// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicingOrdersTableStatusBar should match snapshot 1`] = `
<div
  className="nexus-c-servicing-orders-table-status-bar"
>
  <span
    className="nexus-c-servicing-orders-table-status-bar__description"
  >
    Rows: 
    <span
      className="nexus-c-servicing-orders-table-status-bar__value"
    >
      5
    </span>
  </span>
  <span
    className="nexus-c-servicing-orders-table-status-bar__description"
  >
    Selected: 
    <span
      className="nexus-c-servicing-orders-table-status-bar__value"
    >
      3
    </span>
  </span>
</div>
`;
